website_sale_product_detail_attribute_image/tests/test_website_sale_product_detail_attribute_image.py

```diff
@@ -4,40 +4,49 @@
 
 
 class WebsiteSaleHttpCase(HttpCase):
-    def setUp(self):
-        super().setUp()
-        self.ProductAttribute = self.env["product.attribute"]
-        self.ProductAttributeValue = self.env["product.attribute.value"]
-        self.ProductAttributeLine = self.env["product.template.attribute.line"]
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(
+            context=dict(
+                cls.env.context,
+                mail_create_nolog=True,
+                mail_create_nosubscribe=True,
+                mail_notrack=True,
+                no_reset_password=True,
+                tracking_disable=True,
+            )
```

Comment:
suggestion: create a `DISABLED_MAIL_CONTEXT` variable and use that here:

```suggestion
            context=dict(
                cls.env.context,
                **DISABLED_MAIL_CONTEXT,
            )
```

This way, when fwporting to v16, you can just [import that variable](https://github.com/odoo/odoo/blob/d5c3d47f80d88301e2311c9c1d6eabedca8a562d/odoo/addons/base/tests/common.py#L10-L16) and the diff will be smaller.

The same applies elsewhere.

Reply:
Oh, TIL. @josep-tecnativa please change from now on to do such change. More compact and elegant, and also Odoo can expand this context if needed and we won't need any change in our part.